hedera-mirror-rosetta/app/persistance/postgres/transaction/transaction.go

```diff
@@ -65,14 +71,51 @@ func (transactionStatus) TableName() string {
 }
 
 // The current format used is {payer-ID}-{valid_start_seconds}-{valid_start_nanos}-{node-ID}
-func (t *transaction) constructID() string {
+func (t *transaction) constructIdentifier() string {
 	payerID := constructAccount(t.PayerAccountID)
 	nodeID := constructAccount(t.NodeAccountID)
-	formattedTime := float64(t.ValidStartNS) / float64(1e9)
 
+	validStartBn, tenToNinthBn := new(big.Float).SetInt(big.NewInt(t.ValidStartNS)), big.NewFloat(1e9)
+	formattedTime := validStartBn.Quo(validStartBn, tenToNinthBn)
 	return fmt.Sprintf("%s-%.9f-%s", payerID.FormatToString(), formattedTime, nodeID.FormatToString())
 }
 
+// Populates transaction struct from a given Identifier
+func newTransactionFromIdentifier(identifier string) (*transaction, *rTypes.Error) {
+	inputs := strings.Split(identifier, "-")
+	if len(inputs) != 3 {
+		return nil, errors.Errors[errors.InvalidTransactionIdentifier]
+	}
+
+	payerID, err := types.AccountFromString(inputs[0])
+	if err != nil {
+		return nil, errors.Errors[errors.InvalidTransactionIdentifier]
+	}
+	nodeID, err := types.AccountFromString(inputs[2])
+	if err != nil {
+		return nil, errors.Errors[errors.InvalidTransactionIdentifier]
+	}
+
+	var err1 error
```

Comment:
You do not need to define err1, you can just use := the first time. Also why not reuse the good old `err` variable?

Reply:
1) I cannot use `:` because I am assigning value to the `.PayerAccountID` prop
2) the errors are from different types (one of them is `error` and the other is `rTypes.Error`

Reply:
:(